platform/ios/app/MBXViewController.m

```diff
@@ -709,6 +724,79 @@ - (void)removeParkLayer
     [self.mapView.style removeLayer:parkLayer];
 }
 
+- (void)styleFilteredFill
+{
+    // set style and focus on Texas
+    [self.mapView setStyleURL:[NSURL URLWithString:@"mapbox://styles/justin/citc7ya7w00002jpa0z16cjah"]];
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(31, -100) zoomLevel:3 animated:NO];
+
+    // after slight delay, fill in Texas
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^
+    {
+        MGLFillStyleLayer *statesLayer = (MGLFillStyleLayer *)[self.mapView.style layerWithIdentifier:@"states"];
+
+        // filter
+        statesLayer.predicate = [NSPredicate predicateWithFormat:@"name == 'Texas'"];
+
+        // paint properties
+        statesLayer.fillColor = [UIColor redColor];
+        statesLayer.fillOpacity = @(0.25);
+
+        // layout property to trigger reparse
+        statesLayer.visible = @YES;
+    });
+}
+
+- (void)styleFilteredLines
+{
+    // set style and focus on lower 48
+    [self.mapView setStyleURL:[NSURL URLWithString:@"mapbox://styles/justin/citen6af800652ip1omttobg8"]];
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(40, -97) zoomLevel:5 animated:NO];
+
+    // after slight delay, change styling for all Washington-named counties
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^
+    {
+        MGLLineStyleLayer *countiesLayer = (MGLLineStyleLayer *)[self.mapView.style layerWithIdentifier:@"counties"];
+
+        // filter
+        countiesLayer.predicate = [NSPredicate predicateWithFormat:@"NAME10 == 'Washington'"];
+
+        // paint properties
+        countiesLayer.lineColor = [UIColor redColor];
+        countiesLayer.lineOpacity = @(0.75);
+        countiesLayer.lineWidth = @(5);
+
+        // layout property to trigger reparse
+        countiesLayer.visible = @YES;
+    });
+}
+
+- (void)styleNumericFilteredFills
+{
+    // set style and focus on Texas
+    [self.mapView setStyleURL:[NSURL URLWithString:@"mapbox://styles/justin/citerrmcg00662hogpcv4o3w8"]];
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(40, -97) zoomLevel:5 animated:NO];
+
+    // after slight delay, change styling for all regions 200-299
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^
+    {
+        MGLFillStyleLayer *regionsLayer = (MGLFillStyleLayer *)[self.mapView.style layerWithIdentifier:@"regions"];
+
+        // filter
+        regionsLayer.predicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[
```

Comment:
This can be expressed as a single predicate format string using the `AND` operator.